libraries/chain/webassembly/privileged.cpp

```diff
@@ -155,25 +156,27 @@ namespace eosio { namespace chain { namespace webassembly {
       EOS_ASSERT(!context.trx_context.is_read_only(), wasm_execution_error, "set_proposed_finalizers not allowed in a readonly transaction");
       fc::datastream<const char*> ds( packed_finalizer_set.data(), packed_finalizer_set.size() );
       finalizer_set finset;
-      fc::raw::unpack(ds, finset);
-      vector<finalizer_authority> & finalizers = finset.finalizers;
+      // contract finalizer_set does not include uint32_t generation
+      // struct abi_finalizer_set {
+      //   uint64_t fthreshold
+      //   vector<finalizer_authority> finalizers; }
+      fc::raw::unpack(ds, finset.fthreshold);
+      fc::raw::unpack(ds, finset.finalizers);
+
+      vector<finalizer_authority>& finalizers = finset.finalizers;
 
-      // TODO: check version and increment it or verify correct
       EOS_ASSERT( finalizers.size() <= config::max_finalizers, wasm_execution_error, "Finalizer set exceeds the maximum finalizer count for this chain" );
       EOS_ASSERT( finalizers.size() > 0, wasm_execution_error, "Finalizer set cannot be empty" );
 
       std::set<fc::crypto::blslib::bls_public_key> unique_finalizer_keys;
-#warning REVIEW: Is checking for unique finalizer descriptions at all relevant?
-      std::set<std::string> unique_finalizers;
       uint64_t f_weight_sum = 0;
 
       for (const auto& f: finalizers) {
          f_weight_sum += f.fweight;
          unique_finalizer_keys.insert(f.public_key);
-         unique_finalizers.insert(f.description);
+         EOS_ASSERT( f.description.size() <= config::max_finalizer_description, wasm_execution_error, "Finalizer description greater than 256" );
```

Comment:
You can move this assert to the beginning of this loop body (before `f_weight_sum += f.fweight;`)

Reply:
Done